--- a/src/system.h
+++ b/src/system.h
@@ -79,9 +79,11 @@
 #endif
 
 #ifdef HAVE_LINUX_SYSINFO_H
+#ifndef HAVE_SYS_SYSINFO_H
 #define _LINUX_KERNEL_H
 #include <linux/sysinfo.h>
 #endif
+#endif
 
 #ifdef HAVE_SYS_SYSINFO_H
 #include <sys/sysinfo.h>
@@ -135,6 +137,7 @@
 #include <linux/if_tun.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#undef HAVE_NET_ETHERNET_H
 #ifndef HAVE_SYS_UN_H
 #include <linux/un.h>
 #endif
